### PR TITLE
Don't assign undeclared attributes

### PR DESCRIPTION
**Pull request by jtorreggiani**

### CODE REVIEW

File: spec/fixtures/exampleRelationalResponses.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 export const exampleRelatedToManyResponse = JSON.stringify({
   data: {
     id: '1',
-    type: 'organizations',
+    type: 'todos',
```
Comment (jtorreggiani) [inline spec/fixtures/exampleRelationalResponses.js]: Unclear why we changed this to organizations in the spec and the mocks. Didn't seem to make any different in the specs and was confusing having stuff like this.

```JavaScript
const todo = store.add('organizations', { id: 1, title: 'Buy Milk' }) 
const todo = new Organization()
```

File: src/Model.js [hunk 2]
```diff
@@ -447,12 +447,21 @@ class Model {
...
-
-     extendObservable(this, {
-       ...defaultAttributes,
-       ...initialAttributes
-     })
+     const { defaultAttributes, attributeNames } = this
+     // Make sure to also include id, relationships, and store
+     const names = [...attributeNames, 'id', 'relationships', 'store']
+
+     // NOTE: Discuss behavior or attribute defaults
```
Comment (jtorreggiani) [inline src/Model.js]: If the initial attribute is not present the default value will be used instead.

Comment (jtorreggiani) [reply src/Model.js]: This caused a test to fail and made me wonder if we should think about this behavior. In practice, right now we are not using the default values in many places, but I think it is still worth talking through.

File: spec/integration.spec.js [hunk 3]
```diff
@@ -12,6 +12,7 @@ class Todo extends Model {
   static endpoint = 'todos'
 
   @attribute(String) title = ''
+  @attribute(Object) options
```
Comment (jtorreggiani) [inline spec/integration.spec.js]: Attribute needed to be defined now in order for the "integration" test to work properly.

File: spec/fixtures/exampleRelationalResponses.js (hunk 1 above)
Comment (dashcraft) [reply spec/fixtures/exampleRelationalResponses.js]: This was when i was testing organization relations, it should've been removed, this ended up being a backend issue with pluralizations.

File: spec/integration.spec.js (hunk 3 above)
Comment (dashcraft) [reply spec/integration.spec.js]: Is options going to be required for all models now? Like completions and crop batches?

Comment (jordanstephens) [reply spec/integration.spec.js]: I think this is just because we use `options` down in the `can edit an existing model` test.

**Comment by jordanstephens** [comment]
We should definitely load this into portal and run integrations tests and try to click around and find any other places where we have forgotten to declare `@attributes`. I bet there is at least one more somewhere out there.

**Comment by jordanstephens** [comment]
@jtorreggiani looks like this is stale and unlikely to be merged. can we close this?

**Review by jordanstephens — COMMENTED** [review]
@jtorreggiani looks like this is stale and unlikely to be merged. can we close this?

**Comment by jtorreggiani** [comment]
I will have to spin up on this again. I recall this being something we wanted to do, but having a hard time remember what the specific area this was needed.

**Pull request closed without merging**